kotishell/src/lists/lists_a.c

```diff
@@ -0,0 +1,61 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   lists_a.c                                          :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: elenavoronin <elnvoronin@gmail.com>          +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2023/06/29 16:36:32 by elenavoroni   #+#    #+#                 */
+/*   Updated: 2023/07/14 16:05:30 by evoronin      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "lists.h"
+
+int	li_list_size(t_li_node *list)
+{
+	int		i;
+
+	i = 0;
+	while (list)
+	{
+		i++;
+		list = list->next;
+	}
+	return (i);
+}
+
+t_li_node	*li_new_list(void *data)
+{
+	t_li_node	*node;
+
+	node = mini_malloc(sizeof(t_li_node));
+	if (!node)
+		return (NULL);
+	node->data = data;
+	node->next = NULL;
+	return (node);
+}
+
+void	li_new_stack(t_li_node **list, void *data)
```

Comment:
Make it return status code: 0 if allocation worked, -1 otherwise;

Reply:
fixed in a different way, let's talk about it